### ISSUE TITLE
Offer integration into a mission control system for application autonomy

### ISSUE THREAD

**Issue by SteveMacenski** [ros-planning/navigation2#2758]
Examples:
- A Behavior Tree based system using the `NavigateToPose` higher level primitive to use the Nav2 stack in an autonomy application (e.g. BT.CPP)
- A state machine based system using wrapped clients to Nav2 (e.g. smach, smacc, etc)
- A sequential based system (e.g. @allenh1 )

**Comment by SteveMacenski** [comment]
Yeah, the higher-level autonomy software, which could be one of a number of things depending on the application. Some examples of what I'm thinking are above including a BT-based solution (as you describe), a sequential task autonomy system (get a bunch of primitives of things to do and do them in order), or a scheduler (given a bunch of options and weights / objective functions, pick what to do next)

**Comment by padhupradheep** [comment]
@SteveMacenski I understand this will be a bigger project by itself, are we targeting to start with this anytime this year? Or is it for the next year?

**Comment by jwallace42** [comment]
Next year thing :)

**Comment by mikeferguson** [comment]
Possibly consider integration with RMF instead of building another fleet planner? I've not used it, but https://github.com/open-rmf has some interesting stuff - such as traffic editor GUIs: https://github.com/open-rmf/rmf_traffic_editor
